probe-rs/src/session.rs

```diff
@@ -432,6 +423,111 @@ impl Session {
         Ok(interface)
     }
 
+    fn reattach_arm_interface(
+        interface: &mut Box<dyn ArmProbeInterface>,
+        debug_sequence: &Arc<dyn ArmDebugSequence>,
+    ) -> Result<(), Error> {
+        use crate::DebugProbe;
+
+        // Temporarily swap out the interface for a Dummy
+        let tmp_interface = Box::new(FakeProbe::default())
+            .try_get_arm_interface()
+            .unwrap();
+        let mut tmp_interface = tmp_interface
+            .initialize(DefaultArmSequence::create())
+            .unwrap();
+
+        std::mem::swap(interface, &mut tmp_interface);
```

Comment:
I am a bit confused this works on trait objects where size can differ between the different instantiated objects. What am I missing?

Reply:
Ah, I am stupid, this swaps the entire Box.